vllm/prefix.py

```diff
@@ -44,43 +47,111 @@ def get_length(self) -> int:
     def __hash__(self) -> int:
         return self.hash
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Prefix):
+            return False
+        return self.hash == other.hash
+
     def set_block_table(self, block_table: BlockTable) -> None:
         self.block_table = block_table.copy()
 
 
 class PrefixPool:
-    """Manages all the prompt prefixes.
-
-    NOTE: This feature is experimental and may be replaced with automatic
-        prefix caching in the future.
+    """Manages all the prompt prefixes. If the max_capacity argument is not None,
+    the pool will act as a LRU cache and remove the least recently used prefix once
+    the capacity is reached.
 
     Args:
         block_size: The block size of the executed model.
 
     Attributes:
         prefixes: A list of all the prefixes.
         block_size: The block size of the executed model.
+        max_capacity: The maximum number of prefixes allowed in the pool at any given time. 
+            The default value is None, which means there is no limit. It can only take positive
+            values.
     """
 
-    def __init__(
-        self,
-        block_size: int,
-    ) -> None:
-        # TODO(zhuohan): Add a capacity limit to the prefix pool.
-        self.prefixes: Dict[int, Prefix] = {}
+    def __init__(self,
+                 block_size: int,
+                 max_capacity: Optional[int] = None) -> None:
```

Comment:
Shall we make `max_capacity` as a arg parameter in CacheConfig?

Reply:
@esmeetu I was thinking also as an alternative to make it a parameter of SchedulerConfig, given that PrefixPool is an object of the scheduler.

Reply:
Fixed it here. At the end I added it to CacheConfig:
[9408460](https://github.com/vllm-project/vllm/pull/2511/commits/9408460071488430c6ace7200deb3842738a9805)